<a href="https://colab.research.google.com/github/galihtw04/data-science-2026/blob/master/Pertemuan2_Muhammad_Galih_Abdurrahman_240401010278.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Pertemuan 2
Nama : Muhammad Galih Abdurrahman

NIM : 240401010278

Kelas IF403

In [1]:
# Muat library
import pandas as pd
import numpy as np

# Muat Dataset Titanic (menggunakan Link berbeda karena tidak bisa diakses)
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)
print("Dataset berhasil dimuat!")

Dataset berhasil dimuat!


In [2]:
print(df.shape, df.columns.tolist())
print(df.isnull().sum())

(891, 12) ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64


In [3]:
baris, kolom = df.shape
print(f"Jumlah baris dan kolom dataset Titanic: {baris} baris dan {kolom} kolom")

Jumlah baris dan kolom dataset Titanic: 891 baris dan 12 kolom


In [4]:
missing = df.isnull().sum()
kolom_missing = missing[missing > 0]
print("Kolom yang memiliki nilai kosong (missing):")
for col, jumlah in kolom_missing.items():
    print(f"  - {col}: {jumlah} nilai kosong")

Kolom yang memiliki nilai kosong (missing):
  - Age: 177 nilai kosong
  - Cabin: 687 nilai kosong
  - Embarked: 2 nilai kosong


In [5]:
persen_selamat = df['Survived'].mean() * 100
print(f"Persen penumpang yang selamat (Survived=1): {persen_selamat:.2f}%")

Persen penumpang yang selamat (Survived=1): 38.38%


In [6]:
wk1 = df[(df['Sex']=='female') & (df['Pclass']==1)]
print(wk1['Survived'].mean(), wk1['Age'].mean())

rata_usia = wk1['Age'].mean()
print(f"Rata-rata usia penumpang wanita kelas 1 yang selamat: {rata_usia:.2f} tahun")

0.9680851063829787 34.61176470588235
Rata-rata usia penumpang wanita kelas 1 yang selamat: 34.61 tahun


In [7]:
hasil = df.groupby('Pclass').agg(
    total=('PassengerId','count'),
    selamat=('Survived','mean'),
).round(3)
print(hasil)

kelas_tertinggi = hasil['selamat'].idxmax()
survival_tertinggi = hasil['selamat'].max() * 100
print(f"\nKelas penumpang dengan tingkat keselamatan tertinggi: Kelas {kelas_tertinggi} ({survival_tertinggi:.1f}%)")
print(f"\nHasil groupby('Pclass')['Survived'].mean():")
print(df.groupby('Pclass')['Survived'].mean().round(3))

        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242

Kelas penumpang dengan tingkat keselamatan tertinggi: Kelas 1 (63.0%)

Hasil groupby('Pclass')['Survived'].mean():
Pclass
1    0.630
2    0.473
3    0.242
Name: Survived, dtype: float64


## Kesimpulan

Pada Pertemuan 2 ini, saya mempelajari cara melakukan eksplorasi awal dataset menggunakan Python, khususnya dengan library Pandas dan NumPy. Dataset yang digunakan adalah dataset Titanic yang memiliki 891 baris dan 12 kolom. Dari hasil pengecekan awal, ditemukan beberapa kolom yang memiliki nilai kosong, yaitu kolom Age, Cabin, dan Embarked. Hal ini menunjukkan bahwa sebelum data digunakan untuk analisis lebih lanjut, perlu dilakukan proses pengecekan dan penanganan missing value.

Dari hasil analisis sederhana, diketahui bahwa persentase penumpang yang selamat adalah sekitar 38,38%. Selain itu, analisis terhadap penumpang wanita kelas 1 menunjukkan tingkat keselamatan yang sangat tinggi, yaitu sekitar 96,81%, dengan rata-rata usia sekitar 34,61 tahun. Berdasarkan pengelompokan kelas penumpang, kelas 1 memiliki tingkat keselamatan tertinggi sebesar 63,0%, dibandingkan kelas 2 sebesar 47,3% dan kelas 3 sebesar 24,2%.

Temuan utama dari pertemuan ini adalah bahwa faktor seperti jenis kelamin dan kelas penumpang memiliki hubungan yang cukup kuat terhadap peluang keselamatan penumpang Titanic. Keterbatasan dari analisis ini adalah masih bersifat eksplorasi awal dan belum dilakukan pembersihan data secara mendalam maupun pembuatan model prediksi. Pertanyaan lanjutan yang muncul adalah bagaimana hasil analisis akan berubah jika missing value ditangani dan variabel lain seperti usia, tarif, serta jumlah keluarga ikut dianalisis lebih lanjut.